app/src/containers/MovieDetailsPage/MovieDetailsPage.jsx

```diff
@@ -12,6 +12,8 @@ import getMovieByIdAndSimilarMoviesAsync from './actions';
 
 export class MovieDetailsPage extends React.Component {
   componentDidMount() {
+    if (this.props.movie.id === this.props.match.params.id) return;
```

Comment:
This part fails. Please, update it to:
```javascript
if (Number(this.props.movie.id) === Number(this.props.match.params.id)) return;
```
I've tested your code. This is the only one thing, which is left. I give you 5 but, please, fix this.

Reply:
Good catch, first I got it working, but when fighting with always fresh store on main page I broke that. Fixed in another pr. Big thanks for your help!